### PR TITLE
EGO Gifts

### PR DESCRIPTION
**Pull request by dsmith328**
## About The Pull Request
Adds EGO Gifts for almost every abnormality that we currently have along with the framework to EASILY add more. I made SLOTS for them, which allows admins to see what EGO gifts people have on and eventually set up a UI to Lock the gifts. They replace each other SEAMLESSLY. Almost EVERY SINGLE FUCKING SPRITE was done by the INSANE ButteredToast601. The main change so far is that Mimicry does not provide the healing bonus and Sound of a Star provides a flat justice boost instead of just movement speed.

## Why It's Good For The Game
Little bit of extra stats is just nice. Also it's a step toward LC.

## Changelog
:cl:
add: EGO Gifts for all LC Abnos and Bottle of Tears
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/ego_gifts.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#define MOUTH_2 "Mouth_Slot_2"
+#define CHEEK "Cheek_Slot"
+#define BROOCH "Brooch_Slot"
+#define NECKWEAR "Neckwear_Slot"
+#define LEFTBACK "Left_Back_Slot"
+#define RIGHTBACK "Right_Back_Slot"
+#define HAND_1 "Hand_Slot_1"
+#define HAND_2 "Hand_Slot_2"
+#define SPECIAL "Special_Other_Slot"
+/ego_gifts // Currently Covers most EGO Gift Functions, most others can be done via armors
```
Comment (EgorDinamit) [inline code/modules/mob/living/carbon/human/ego_gifts.dm:15]:
```suggestion
/datum/ego_gift // Currently Covers most EGO Gift Functions, most others can be done via armors
```
Don't forget to change it everywhere else.

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 2]
```diff
@@ -82,3 +82,6 @@
 
 	/// Boolean for sanity points loss effects
 	var/sanity_lost = FALSE
+
+	/// List of EGO Gifts
+	var/list/ego_gifts/ego_gift_list = list()
```
Comment (EgorDinamit) [inline code/modules/mob/living/carbon/human/human_defines.dm:87]:
```suggestion
	var/list/ego_gift_list = list()
```
List with pre-defined types will not be supported sooner or later by Byond, from what I've heard.

File: code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm [hunk 3]
```diff
@@ -99,6 +99,8 @@
 		datum_reference.qliphoth_change(-1)
 	if(user.sanity_lost)
 		datum_reference.qliphoth_change(-1)
+	if(prob(1))
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm:102]: All of it should be moved to variables and _abnormality.dm

File: code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm [hunk 4]
```diff
@@ -108,3 +110,9 @@
 	forceMove(T)
 	BluePulse()
 	return
+
+/ego_gifts/star
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm:114]: I'd say all ego gifts should be in a separate folder.

File: code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm [hunk 5]
```diff
@@ -58,41 +53,123 @@
...
-					user.adjust_attribute_buff(FORTITUDE_ATTRIBUTE, -10)
-					user.adjust_attribute_buff(JUSTICE_ATTRIBUTE, 5)
-					user.add_overlay(mutable_appearance(buff_icon, "foolish", -ABOVE_MOB_LAYER))
-					to_chat(user, "<span class='userdanger'>You are a God among men!</span>")
-					foolish_samurai_warrior[user] = 4
-					return
-				if(4) // You can progress no further down this fool-hardy path
-					return
-		else // Give Courage
+		if (src.icon_state == "megalovania")
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm:56]: I don't know if I should hate you for this or not.

Comment (dsmith328) [reply code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm:56]: You should.

File: code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm (hunk 4 above)
Comment (dsmith328) [reply code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm:114]: Done~

File: code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm (hunk 3 above)
Comment (dsmith328) [reply code/modules/mob/living/simple_animal/abnormality/aleph/blue_star.dm:102]: Done~

File: code/modules/mob/living/carbon/human/human_defines.dm (hunk 2 above)
Comment (dsmith328) [reply code/modules/mob/living/carbon/human/human_defines.dm]: Linters require it to be a pre-defined type list.

File: code/modules/mob/living/carbon/human/ego_gifts.dm (hunk 1 above)
Comment (dsmith328) [reply code/modules/mob/living/carbon/human/ego_gifts.dm]: Done digity done

**Comment by dsmith328** [comment]
Crumbling Armor is staying as an outlier as it is overall different from the other Abnormalities' gifts. Attempting to rewrite the entirety of the application code or doing at the very least major changes on crumbling armor's side just straight up isn't worth it for a singular abnormality to conform to the rest.

**Comment by EgorDinamit** [comment]
`code\modules\mob\living\simple_animal\abnormality\teth\crumbling_armor.dm:41:error: ego_gift_list: undefined var`
You forgot "user" here.

**Pull request merged**